lib/iris/cube.py

```diff
@@ -3131,6 +3084,10 @@ def __eq__(self, other):
         if isinstance(other, Cube):
             result = self.metadata == other.metadata
 
+            # check the cube fill-value.
+            if result:
+                result = self.fill_value == other.fill_value
```

Comment:
@dkillick I raised issue https://github.com/SciTools/iris/issues/2497 so that we don't sweep this one under the carpet.